internal/rollout/device_selection/batch_sequence_selector.go

```diff
@@ -0,0 +1,462 @@
+package device_selection
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/store/selector"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultSuccessThreshold = 90
+	Unknown                 = "Unknown"
+	ApplyingUpdate          = "ApplyingUpdate"
+	Rebooting               = "Rebooting"
+	ActivatingConfig        = "ActivatingConfig"
+	RollingBack             = "RollingBack"
+	Error                   = "Error"
+)
+
+func newBatchSequenceSelector(sequence api.BatchSequence, defaultUpdateTimeout *time.Duration, store store.Store, orgId uuid.UUID, fleet *api.Fleet, templateVersionName string, log logrus.FieldLogger) RolloutDeviceSelector {
+	return &batchSequenceSelector{
+		BatchSequence:        sequence,
+		store:                store,
+		orgId:                orgId,
+		fleetName:            lo.FromPtr(fleet.Metadata.Name),
+		fleet:                fleet,
+		templateVersionName:  templateVersionName,
+		defaultUpdateTimeout: defaultUpdateTimeout,
+		log:                  log,
+	}
+}
+
+type batchSequenceSelector struct {
+	api.BatchSequence
+	store                store.Store
+	orgId                uuid.UUID
+	fleet                *api.Fleet
+	fleetName            string
+	templateVersionName  string
+	defaultUpdateTimeout *time.Duration
+	log                  logrus.FieldLogger
+}
+
+func (b *batchSequenceSelector) getAnnotation(annotation string) (string, bool) {
```

Comment:
Why not move to a util file, maybe api/v1alpha1/util.go?

Reply:
OK